Source/Database.h

```diff
@@ -0,0 +1,188 @@
+    String textSearch (const String & searchString)
+    {
+        String query = String ("{ \"selector\": { \"$text\": \"" + searchString + "\" } }");
+        //DBG (query);
+        
+        // Cloudant's text query facility
+        post ("/_find", query);
+        
+        checkStatus();
+
+        DBG (queryResult);
+        
+        return queryResult;
+    }
+    
+    String getModulesInRepo (const String & repoNameString)
+    {
+        
+        // Build query string to find module or module set based on shortname
+        String query = String ("{ \"selector\": { \"shortname\": \"" + repoNameString + "\" } }");
+        //DBG (query);
+        
+        // Cloudant's text query facility
+        post ("/_find", query);
+        
+        checkStatus();
+        
+        DBG (queryResult);
+        
+        return queryResult;
+    }
+    
+    /** Get modules matching a given name.
+     *  This calls a pre-defined 'view' which is a javascript map function defined on the database.
+     *
+     *  For reference the map function, as of the last update to this file:
+     *      function (doc) {
+     *          if (doc.module) {
+     *              if (Array.isArray(doc.module)) {
+     *                  doc.module.forEach (function (module) {
+     *                      emit (module.name, {repo: {shortname: doc.shortname, path: doc.path, source: doc.source}, module: module});
+     *                  });
+     *              } else {
+     *                  emit (doc.module.name, {repo: {shortname: doc.shortname, path: doc.path, source: doc.source}, module: doc.module});
+     *              }
+     *          }
+     *      }
+     */
+    String getModulesByName (const String & moduleNameString)
+    {
+        // Call predefined "module-name" view.
+        
+        get ("_design/module-views/_view/module-name?key=%22" + moduleNameString + "%22");
+        
+        checkStatus();
+
+        return queryResult;
+    }
+    
+    URL &getURL()
+    {
+        return url;
+    }
+
+private:
+    
+    /** Check for error status, print message and return false if error status found
+     */
+    bool checkStatus()
+    {
+        if (status < 200 || status > 300)
+        {
+            std::cerr << "fatal: returned status " << status << ":" << std::endl;
+            std::cerr << queryResult;
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     create a GET HTTP request to an endpoint based on the stored URL
+     and store the result
+     */
+    void get (const String & endpoint)
+    {
+        const ScopedPointer<InputStream> in(url.getChildURL (endpoint).createInputStream (false, nullptr, nullptr, headers, 0, &responseHeaders, &status));
+        
+        //for (auto key : responseHeaders.getAllKeys())
+        //{
+        //    DBG (key << ": " << responseHeaders.getValue(key, "n/a"));
+        //}
+        //DBG (status);
+
+        queryResult = in->readEntireStreamAsString();

```

Comment:
Try uncommenting the above DBG statements and see what you get, if you're still having trouble.